src/components/dashboard/container/widget_container.gd

```diff
@@ -482,6 +484,13 @@ func get_data_sources_data() -> Array:
 onready var query_warning_title = $QueryWarning/VBox/PanelContainer/VBox/QueryStatusTitle
 func _on_data_source_query_status(_type:int=0, _title:="Widget Error", _message:=""):
 	if _type == OK:
+		var querying = false
+		for datasource in data_sources:
+			if datasource.is_executing_query():
+				querying = true
+				break
+		if not querying:
+			loading_overlay.visible = false
 		return
 	query_warning_title.visible = _title != ""
```

Comment:
If the returning error (`_type`) is not `OK`, the loading_overlay will still be visible when the warning is showing?

Reply:
Please also check on other places if the loading overlay is hidden when things do not go the happy path 😵